internal/commands/result.go

```diff
@@ -683,21 +714,27 @@ func CreateScanReport(
 	targetPath string,
 	params map[string]string,
 ) error {
-	err := createDirectory(targetPath)
+	reportList := strings.Split(reportTypes, ",")
+	results := &wrappers.ScanResultsCollection{}
+
+	summary, err := summaryReport(scan, policyResponseModel, risksOverviewWrapper, resultsWrapper)
 	if err != nil {
 		return err
 	}
-	results, err := ReadResults(resultsWrapper, scan, params)
+
+	err = createDirectory(targetPath)
 	if err != nil {
 		return err
 	}
 
-	summary, err := SummaryReport(results, scan, policyResponseModel, risksOverviewWrapper, resultsWrapper)
-	if err != nil {
-		return err
+	isResultsNeeded := verifyFormatsByReportList(reportList, printer.FormatJSON, printer.FormatSarif, printer.FormatSonar)
```

Comment:
Consider having some static lists of summary report types and result report types in the printer package.